packages/start-plugin/src/nitro/plugins/dev-server-plugin.ts

```diff
@@ -0,0 +1,124 @@
+// SSR dev server, middleware and error page source modified from
+// https://github.com/solidjs/solid-start/blob/main/packages/start/dev/server.js
+
+import { createEvent, sendWebResponse } from 'h3'
+
+import { registerDevServerMiddleware } from '../utils/register-dev-middleware.js'
+import type { Connect, Plugin, ViteDevServer } from 'vite'
+import type { TanStackStartOutputConfig } from '../../schema.js'
+
+export function devServerPlugin(options: TanStackStartOutputConfig): Plugin {
+  // let config: UserConfig
+  let isTest = false
+
+  return {
+    name: 'startjs-dev-ssr-plugin',
+    config(userConfig, { mode }) {
+      // config = userConfig
+      isTest = isTest ? isTest : mode === 'test'
+      return {
+        resolve: {
+          alias: {
+            '~start/ssr-entry': options.ssrEntryPath,
+          },
+        },
+      }
+    },
+    configureServer(viteServer) {
+      if (isTest) {
+        return
+      }
+
+      return () => {
+        remove_html_middlewares(viteServer.middlewares)
+        registerDevServerMiddleware(options.root, viteServer)
+
+        viteServer.middlewares.use(async (req, res) => {
+          const template = await viteServer.transformIndexHtml(
+            req.originalUrl as string,
+            `<html>
+              <head></head>
+              <body></body>
+            </html>`,
+          )
+
+          console.log('template, maybe use?', template)
```

Comment:
If we can, we probably should use this, since it'd have the necessary `@vitejs/plugin-react` bindings baked into it. So we can do away with this workaround and just let the React plugin do its thing.
https://github.com/TanStack/router/blob/4a9d7717dd32f507bc146adafeb7939bd479f95b/packages/start-router-manifest/src/index.ts#L34-L39

Reply:
For solid, I had to remove this section. It's the only five lines making `start-router-manifest` non-agnostic. Alternatively, if the react-refresh could be added conditioned on the vite plugin target (react|solid).